### DISCUSSION TITLE
Destination Set Filter vs Filtering on Destinations

### DISCUSSION THREAD

**Discussion by EthanKC** [nextgenhealthcare/connect#5097] [Q&A] [answered]
I'm pretty new to Mirth Connect, and still figuring everything out. I already understand how filters work, but just today I discovered the Destination Set Filter Transformer. Am I missing something, or does this basically make the Destination level filtering obsolete? Is there something the Destination level filters can do that can't be done with the Destination Set Filter Transformer at the Source level?

**Comment by pacmano1** [comment]
Destination set filtering, i.e.:

![image](https://user-images.githubusercontent.com/44065187/158268525-54b4b76b-bfc8-4a0d-9cb1-f7f220ea1be8.png)


whereby you remove destinations based on stuff determined on the source side is great.   You can only use those on the source side. 

You may not be able to determine everything in the source side.  

For example, destination 1 might call an API and you may want destination 2 to process only if something determined in the response from destination 1.   In that example you would have a filter on destination 2 based on criteria from destination 1.

**Reply by EthanKC** [reply]
Ok, so if I’m just using Mirth for relatively simple HL7 routing and transformation, they basically do the same thing?

**Reply by jonbartels** [reply]
The end result is the same but the intermediate process is different.

A filter is a destination saying "Should I handle this message or not"
A destination set call is a source transformer script saying "Destination A and B should do this but not C". It is centralized to some extent.

A filter is more visible, it shows the "filtered" statistic and shows what destinations accepted or filtered a message.
A destination set call does save on storage in that with no destination in the destination chain, then no messages are created for that destination. 

Pro-tip - If you decide to use `destinationSet.remove` remember that it returns a boolean. If it returns false you likely have a programming error like a mis-typed destination name. Always check its return value.

**Comment by pacmano1** [comment, marked as answer]
Answered in thread.
